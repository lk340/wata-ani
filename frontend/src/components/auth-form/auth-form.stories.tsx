import * as React from "react";
import * as Reach from "@reach/router";

import * as Wrappers from "@/wrappers";

import { AuthForm } from "./auth-form";

type Story = {
	component: React.ReactNode;
	title: string;
	excludeStories: RegExp;
};

const story: Story = {
	component: AuthForm,
	title: "AuthForm",
	excludeStories: /.*Data$/,
};

export default story;

export const authFormData = {};

export const Register = () => {
	return (
		<Reach.LocationProvider>
			<Wrappers.Layout>
				<AuthForm formType="Registration" submitText="Register" />
			</Wrappers.Layout>
		</Reach.LocationProvider>
	);
};

export const SignIn = () => {
	return (
		<Reach.LocationProvider>
			<Wrappers.Layout>
				<AuthForm formType="Sign In" submitText="Sign In" />
			</Wrappers.Layout>
		</Reach.LocationProvider>
	);
};
