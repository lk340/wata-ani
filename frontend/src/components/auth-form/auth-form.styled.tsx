import styled from "styled-components";
import { animated } from "react-spring";
import * as Gatsby from "gatsby";

import * as Constants from "@/utils/style/constants";
import * as Snippets from "@/utils/style/snippets";
import * as Colors from "@/utils/style/colors";

import logoJapanese from "@/images/logo/japanese.svg";

/**
 * Auth Form
 * Logo, Title, & Inputs
 * Submit Button & Redirect
 * Submit Button
 * Redirect
 */

// ================= //
// ↓↓↓ Auth Form ↓↓↓ //
// ================= //

export const AuthForm = styled("form")`
	${Snippets.grid(1, "auto", 40)};
	width: 400px;

	@media (max-width: ${Constants.breakpoints.mobile}) {
		padding: 0px ${Constants.sidePaddings.mobile}px;
		width: 100%;
	}

	@media (max-height: 620px) {
		grid-gap: 20px;
	}
`;

// ============================= //
// ↓↓↓ Logo, Title, & Inputs ↓↓↓ //
// ============================= //

export const AuthFormLogoTitleInputs = styled("div")`
	${Snippets.grid(1, "auto", 30, "auto", "auto")};
`;

export const AuthFormLogo = styled("img").attrs(() => ({
	src: logoJapanese,
	alt: "image",
}))`
	${Snippets.square("70px")};
	margin: 0px auto;
	border-radius: ${Constants.borderRadius.components.authForm.logo};

	@media (max-height: 740px) {
		display: none;
	}
`;

export const AuthFormTitle = styled(animated.h1)`
	${Snippets.clearSpacing()};
	font-size: ${Constants.fontSizes.components.authForm.title};
	text-align: center;
`;

export const AuthFormInputs = styled("div")`
	${Snippets.grid(1, "auto", 20)};
`;

// ================================ //
// ↓↓↓ Submit Button & Redirect ↓↓↓ //
// ================================ //

export const AuthFormSubmitRedirect = styled("div")`
	${Snippets.grid(1, "auto", 20)};
`;

// ===================== //
// ↓↓↓ Submit Button ↓↓↓ //
// ===================== //

export const AuthFormSubmitButton = styled("button").attrs(() => ({
	type: "submit",
}))`
	${Snippets.flexRowCenter()};
	height: 50px;
	color: ${Colors.NEUTRALS.white_100};
	background-color: ${Colors.PRIMARY_100};
	border: none;
	border-radius: ${Constants.borderRadius.components.authForm.input};
	font-size: ${Constants.fontSizes.components.authForm.submit};
	font-weight: bold;
	outline: none;
	cursor: pointer;
`;

// ================ //
// ↓↓↓ Redirect ↓↓↓ //
// ================ //

export const AuthFormRedirect = styled(animated.p)`
	${Snippets.clearSpacing()};
	color: ${Constants.theme.text};
	font-size: ${Constants.fontSizes.components.authForm.redirect};
`;

type RedirectLink = { form_type: "Registration" | "Sign In" };

export const AuthFormRedirectLink = styled(Gatsby.Link).attrs((props: RedirectLink) => ({
	to: props.form_type === "Registration" ? "/sign-in" : "/registration",
}))<RedirectLink>`
	${Snippets.clearAnchor()};
	display: inline;
	color: ${Colors.PRIMARY_100};
	font-weight: bold;
	font-size: ${Constants.fontSizes.components.authForm.redirect};
`;
