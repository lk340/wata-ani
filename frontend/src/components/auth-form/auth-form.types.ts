export type FormType = "Registration" | "Sign In";
