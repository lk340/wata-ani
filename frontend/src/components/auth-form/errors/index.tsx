import * as React from "react";
import * as ReactRedux from "react-redux";
import * as UUID from "uuid";

import * as Styled from "./errors.styled";
import * as Springs from "./errors.springs";

export const Errors = () => {
	const animateText = Springs.text();

	const errors = ReactRedux.useSelector((state) => state.errors.session);

	const errorMessages = errors.map((error: string) => {
		return <Styled.ErrorsText key={UUID.v4()}>{error}</Styled.ErrorsText>;
	});

	return <Styled.Errors style={animateText}>{errorMessages}</Styled.Errors>;
};
