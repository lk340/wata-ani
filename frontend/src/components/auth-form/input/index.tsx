import * as React from "react";

import * as Context from "@/context";

import * as AuthTypes from "../auth-form.types";
import * as Styled from "./input.styled";
import * as Springs from "./input.springs";

type InputProps = {
	formType: AuthTypes.FormType;
	inputType: "Username Or Email" | "Username" | "Email" | "Password" | "Confirm Password";
	onChange: React.ChangeEventHandler;
};

export const Input = (props: InputProps) => {
	const { formType, inputType, onChange } = props;

	const { authForm } = Context.AuthForm.useAuthFormContext();
	const { windows } = Context.Windows.useWindowsContext();

	// Hides or shows certain input fields depending on whether we're dealing
	// with a registration form or a sign in form.
	const display =
		formType === "Registration" && inputType === "Username Or Email"
			? "none"
			: formType === "Sign In" && inputType === "Username"
			? "none"
			: formType === "Sign In" && inputType === "Email"
			? "none"
			: formType === "Sign In" && inputType === "Confirm Password"
			? "none"
			: "grid";

	// Shows the icon that corresponds with the specific input field type.
	// 	e.g. Username or Email field, Username input field, Email input field,
	//	Password input field, Confirm Password input field.
	let icon;
	switch (inputType) {
		case "Username Or Email":
			icon = <Styled.InputIconUsername />;
			break;
		case "Username":
			icon = <Styled.InputIconUsername />;
			break;
		case "Email":
			icon = <Styled.InputIconEmail />;
			break;
		default:
			icon = <Styled.InputIconPassword />;
			break;
	}

	// Renders specific input field depending on input field type.
	let field;
	switch (inputType) {
		case "Username Or Email":
			field = (
				<Styled.InputUsernameOrEmail
					onChange={onChange}
					form_type={formType}
					window_height={windows.state.height}
				/>
			);
			break;
		case "Username":
			field = (
				<Styled.InputUsername
					onChange={onChange}
					form_type={formType}
					window_height={windows.state.height}
				/>
			);
			break;
		case "Email":
			field = (
				<Styled.InputEmail
					onChange={onChange}
					form_type={formType}
					window_height={windows.state.height}
				/>
			);
			break;
		case "Password":
			// "Password" & "Confirm Password" case
			field = (
				<Styled.InputPassword
					onChange={onChange}
					reveal_password={authForm.state.revealPassword}
					window_height={windows.state.height}
				/>
			);
			break;
		default:
			field = (
				<Styled.InputPasswordConfirmation
					onChange={onChange}
					reveal_password={authForm.state.revealPassword}
					window_height={windows.state.height}
					form_type={formType}
				/>
			);
			break;
	}

	return (
		<Styled.Input display={display}>
			<Styled.InputTitle>*{inputType}</Styled.InputTitle>
			<Styled.InputField>
				{icon}
				{field}

				{/* Eye icons for showing or hiding typed-in password. */}
				<Styled.InputFieldPasswordRevealIcons
					onClick={authForm.setters.toggleRevealPassword}
					input_type={inputType}
				>
					<Styled.InputIconPasswordHide
						style={Springs.togglePasswordHideIcon(authForm.state.revealPassword)}
					/>
					<Styled.InputIconPasswordShow
						style={Springs.togglePasswordShowIcon(authForm.state.revealPassword)}
					/>
				</Styled.InputFieldPasswordRevealIcons>
			</Styled.InputField>
		</Styled.Input>
	);
};
