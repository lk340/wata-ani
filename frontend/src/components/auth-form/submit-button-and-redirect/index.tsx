import * as React from "react";

import * as Animations from "@/utils/style/animations";

import * as Styled from "./submit-button-and-redirect.styled";
import { Props } from "../auth-form";

export const SubmitButtonAndRedirect = (props: Props) => {
	const { formType, submitText } = props;

	const animateRedirect = Animations.text();

	return (
		<Styled.SubmitButtonAndRedirect>
			{/* Submit Button */}
			<Styled.SubmitButton>{submitText}</Styled.SubmitButton>

			{/* Redirect */}
			<Styled.Redirect style={animateRedirect}>
				{formType === "Registration" ? "Already a member?" : "Need an account?"}&nbsp;
				{/* Link */}
				<Styled.RedirectLink form_type={formType}>
					{formType === "Registration" ? "Sign In" : "Register"}
				</Styled.RedirectLink>
			</Styled.Redirect>
		</Styled.SubmitButtonAndRedirect>
	);
};
