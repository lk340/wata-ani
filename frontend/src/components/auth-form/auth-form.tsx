import * as React from "react";

import * as Context from "@/context";
import * as Animations from "@/utils/style/animations";

import * as Types from "./auth-form.types";
import * as Styled from "./auth-form.styled";
import { Input } from "./input";

type Props = {
	formType: Types.FormType;
	submitText: "Register" | "Sign In";
};

export const AuthForm = (props: Props) => {
	const { formType, submitText } = props;

	const { authForm } = Context.AuthForm.useAuthFormContext();
	const { theme } = Context.Theme.useThemeContext();

	const animateTitle = Animations.text(theme.state.mode);

	return (
		<Styled.AuthForm
			onSubmit={(event: React.FormEvent<HTMLFormElement>) =>
				authForm.handlers.handleSubmit(event, formType)
			}
		>
			{/* Logo, Title, & Inputs */}
			<Styled.AuthFormLogoTitleInputs>
				{/* Logo */}
				<Styled.AuthFormLogo />
				{/* Title */}
				<Styled.AuthFormTitle style={animateTitle}>{formType}</Styled.AuthFormTitle>
				{/* Inputs */}
				<InputFields formType={formType} submitText={submitText} />
			</Styled.AuthFormLogoTitleInputs>

			{/* Submit Button & Redirect */}
			<SubmitButtonAndRedirect formType={formType} submitText={submitText} />
		</Styled.AuthForm>
	);
};

// ==================== //
// ↓↓↓ Input Fields ↓↓↓ //
// ==================== //

const InputFields = (props: Props) => {
	const { formType } = props;

	const { authForm } = Context.AuthForm.useAuthFormContext();

	return (
		<Styled.AuthFormInputs>
			<Input
				onChange={authForm.handlers.handleUsernameOrEmailChange}
				formType={formType}
				inputType="Username Or Email"
			/>

			<Input
				onChange={authForm.handlers.handleUsernameChange}
				formType={formType}
				inputType="Username"
			/>
			<Input
				onChange={authForm.handlers.handleEmailChange}
				formType={formType}
				inputType="Email"
			/>
			<Input
				onChange={authForm.handlers.handlePasswordChange}
				formType={formType}
				inputType="Password"
			/>
			<Input
				onChange={authForm.handlers.handlePasswordConfirmationChange}
				formType={formType}
				inputType="Confirm Password"
			/>
		</Styled.AuthFormInputs>
	);
};

// ===================================== //
// ↓↓↓ Submit Button & Redirect Link ↓↓↓ //
// ===================================== //

const SubmitButtonAndRedirect = (props: Props) => {
	const { formType, submitText } = props;

	const { theme } = Context.Theme.useThemeContext();

	const animateRedirect = Animations.text(theme.state.mode);

	return (
		<Styled.AuthFormSubmitRedirect>
			{/* Submit Button */}
			<Styled.AuthFormSubmitButton>{submitText}</Styled.AuthFormSubmitButton>

			{/* Redirect */}
			<Styled.AuthFormRedirect style={animateRedirect}>
				{formType === "Registration" ? "Already a member?" : "Need an account?"}&nbsp;
				<Styled.AuthFormRedirectLink form_type={formType}>
					{formType === "Registration" ? "Sign In" : "Register"}
				</Styled.AuthFormRedirectLink>
			</Styled.AuthFormRedirect>
		</Styled.AuthFormSubmitRedirect>
	);
};
