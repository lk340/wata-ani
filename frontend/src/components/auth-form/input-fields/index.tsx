import * as React from "react";

import * as Context from "@/context";

import * as Styled from "./input-fields.styled";

import { FormType } from "../auth-form";
import { Input } from "../input";

type Props = { formType: FormType };

export const InputFields = (props: Props) => {
	const { formType } = props;

	const { authForm } = Context.AuthForm.useAuthFormContext();

	return (
		<Styled.InputFields>
			{/* Username Or Email */}
			<Input
				onChange={authForm.handlers.handleUsernameOrEmailChange}
				formType={formType}
				inputType="Username Or Email"
			/>

			{/* Username */}
			<Input
				onChange={authForm.handlers.handleUsernameChange}
				formType={formType}
				inputType="Username"
			/>

			{/* Email */}
			<Input
				onChange={authForm.handlers.handleEmailChange}
				formType={formType}
				inputType="Email"
			/>

			{/* Password */}
			<Input
				onChange={authForm.handlers.handlePasswordChange}
				formType={formType}
				inputType="Password"
			/>

			{/* Confirm Password */}
			<Input
				onChange={authForm.handlers.handlePasswordConfirmationChange}
				formType={formType}
				inputType="Confirm Password"
			/>
		</Styled.InputFields>
	);
};
