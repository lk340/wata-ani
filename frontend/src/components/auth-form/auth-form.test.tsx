import * as React from "react";
import * as Testing from "@testing-library/react";
import axios from "axios";

import * as Tests from "@/utils/tests";

import { AuthForm } from "./auth-form";

jest.mock("axios");

describe("AuthForm Component", () => {
	it("matches the snapshot", () => {
		const { asFragment } = Tests.renderWithContext(
			<AuthForm formType="Registration" submitText="Register" />,
		);

		expect(
			asFragment(<AuthForm formType="Registration" submitText="Register" />),
		).toMatchSnapshot();
	});

	describe("Sign In Form Type", () => {
		it("renders the logo", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Sign In" submitText="Sign In" />,
			);

			const logo = authForm.getByAltText(/auth form component logo image/);
			expect(logo).not.toBe(null);
		});

		it("renders the form title", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Sign In" submitText="Sign In" />,
			);

			const title = authForm.getByRole("heading", { name: /Sign In/ });
			expect(title).toHaveTextContent(/Sign In/);
		});

		it("renders the Username Or Email field", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Sign In" submitText="Sign In" />,
			);

			const usernameOrEmailField = authForm.getByPlaceholderText(/Username or Email/);
			expect(usernameOrEmailField).not.toBe(null);
		});

		it("renders the Password field", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Sign In" submitText="Sign In" />,
			);

			const passwordField = authForm.getByPlaceholderText(/\*Password/);
			expect(passwordField).not.toBe(null);
		});

		it("renders error messages on erroneous user input", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Sign In" submitText="Sign In" />,
			);

			// const usernameOrEmailField = authForm.getByPlaceholderText(/Username or Email/);
			// expect(usernameOrEmailField).not.toBe(null);

			// const passwordField = authForm.getByPlaceholderText(/\*Password/);
			// expect(passwordField).not.toBe(null);

			// const submitButton = authForm.getByRole("button", { name: /Sign In/ });
			// expect(submitButton).not.toBe(null);

			// Testing.fireEvent.change(usernameOrEmailField, {
			// 	target: { value: "foobar" },
			// });

			// Testing.fireEvent.change(passwordField, {
			// 	target: { value: "justfoothebar" },
			// });

			// Testing.fireEvent.click(submitButton);

			// const errorMessage = authForm.getByText(
			// 	/Unable to log in with provided credentials/,
			// );
			// expect(errorMessage).not.toBe(null);
		});
	});

	describe("Registration Form Type", () => {
		it("renders the logo", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Registration" submitText="Register" />,
			);

			const logo = authForm.getByAltText(/auth form component logo image/);
			expect(logo).not.toBe(null);
		});

		it("renders the form title", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Registration" submitText="Register" />,
			);

			const title = authForm.getByRole("heading", { name: /Registration/ });
			expect(title).toHaveTextContent(/Registration/);
		});

		it("renders the Username field", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Registration" submitText="Register" />,
			);

			const usernameField = authForm.getByTestId(
				/auth form component username input field/,
			);
			expect(usernameField).not.toBe(null);
		});

		it("renders the Email field", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Registration" submitText="Register" />,
			);

			const emailField = authForm.getByPlaceholderText(/\*Email/);
			expect(emailField).not.toBe(null);
		});

		it("renders the Password field", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Registration" submitText="Register" />,
			);

			const passwordField = authForm.getByPlaceholderText(/\*Password/);
			expect(passwordField).not.toBe(null);
		});

		it("renders the Confirm Password field", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Registration" submitText="Register" />,
			);

			const confirmPasswordField = authForm.getByPlaceholderText(/\*Confirm Password/);
			expect(confirmPasswordField).not.toBe(null);
		});

		it("renders error messages on erroneous user input", () => {
			const authForm = Tests.renderWithContext(
				<AuthForm formType="Registration" submitText="Register" />,
			);
		});
	});
});
