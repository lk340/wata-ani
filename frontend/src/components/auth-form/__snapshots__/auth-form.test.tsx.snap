// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm Component matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(1,auto);
  grid-gap: 40px;
  justify-items: auto;
  -webkit-align-items: auto;
  -webkit-box-align: auto;
  -ms-flex-align: auto;
  align-items: auto;
  width: 400px;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(1,auto);
  grid-gap: 30px;
  justify-items: auto;
  -webkit-align-items: auto;
  -webkit-box-align: auto;
  -ms-flex-align: auto;
  align-items: auto;
}

.c2 {
  width: 70px;
  height: 70px;
  margin: 0px auto;
  border-radius: 0.375rem;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 1.5rem;
  text-align: center;
}

.c4 {
  display: grid;
  grid-template-columns: repeat(1,auto);
  grid-gap: 20px;
  justify-items: auto;
  -webkit-align-items: auto;
  -webkit-box-align: auto;
  -ms-flex-align: auto;
  align-items: auto;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(1,auto);
  grid-gap: 10px;
  justify-items: auto;
  -webkit-align-items: auto;
  -webkit-box-align: auto;
  -ms-flex-align: auto;
  align-items: auto;
  display: none;
}

.c14 {
  display: grid;
  grid-template-columns: repeat(1,auto);
  grid-gap: 10px;
  justify-items: auto;
  -webkit-align-items: auto;
  -webkit-box-align: auto;
  -ms-flex-align: auto;
  align-items: auto;
  display: grid;
}

.c6 {
  margin: 0;
  padding: 0;
  color: #6BA6FF;
  font-size: 0.875rem;
  font-weight: bold;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: auto;
  -webkit-justify-content: auto;
  -ms-flex-pack: auto;
  justify-content: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50px;
  padding-left: 20px;
  border-radius: 0.3125rem;
}

.c8 {
  width: 20px;
  height: 20px;
}

.c16 {
  width: 20px;
  height: 20px;
}

.c18 {
  width: 20px;
  height: 20px;
}

.c10 {
  display: none;
  padding-right: 20px;
  cursor: pointer;
}

.c20 {
  display: block;
  padding-right: 20px;
  cursor: pointer;
}

.c11 {
  width: 20px;
  height: 20px;
}

.c12 {
  width: 20px;
  height: 20px;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  margin: 0px 20px;
  background-color: rgba(0,0,0,0);
  border: none;
  outline: none;
}

.c9::-webkit-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c9::-moz-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c9:-ms-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c9::placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c15 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  margin: 0px 20px;
  background-color: rgba(0,0,0,0);
  border: none;
  outline: none;
}

.c15::-webkit-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c15::-moz-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c15:-ms-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c15::placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c17 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  margin: 0px 20px;
  background-color: rgba(0,0,0,0);
  border: none;
  outline: none;
}

.c17::-webkit-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c17::-moz-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c17:-ms-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c17::placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c19 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  margin: 0px 20px;
  background-color: rgba(0,0,0,0);
  border: none;
  outline: none;
}

.c19::-webkit-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c19::-moz-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c19:-ms-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c19::placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c21 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  margin: 0px 20px;
  background-color: rgba(0,0,0,0);
  border: none;
  outline: none;
}

.c21::-webkit-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c21::-moz-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c21:-ms-input-placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c21::placeholder {
  color: #C4C4C4;
  font-size: 0.75rem;
  font-style: italic;
}

.c13 {
  margin: 0;
  padding: 0;
  display: none;
  color: #C32263;
  font-size: 0.75rem;
}

.c22 {
  display: grid;
  grid-template-columns: repeat(1,auto);
  grid-gap: 20px;
  justify-items: auto;
  -webkit-align-items: auto;
  -webkit-box-align: auto;
  -ms-flex-align: auto;
  align-items: auto;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50px;
  color: #FFFFFF;
  background-color: #6BA6FF;
  border: none;
  border-radius: 0.3125rem;
  font-size: 1rem;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c24 {
  margin: 0;
  padding: 0;
  color: light:#212121;
  dark: #F2F2F2;
  font-size: 0.875rem;
}

.c25 {
  display: inherit;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  display: inline;
  color: #6BA6FF;
  font-weight: bold;
  font-size: 0.875rem;
}

@media (max-width:450) {
  .c0 {
    padding: 0px 30px;
    width: 100%;
  }
}

@media (max-height:620px) {
  .c0 {
    grid-gap: 20px;
  }
}

@media (max-height:740px) {
  .c2 {
    display: none;
  }
}

@media (max-height:620px) {
  .c6 {
    display: none;
  }
}

<form
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="auth form component logo image"
        class="c2"
        src="[object Object]"
      />
      <h1
        class="c3"
        style="color: rgb(242, 242, 242);"
      >
        Registration
      </h1>
      <div
        class="c4"
      >
        <div
          class="c5"
          display="none"
        >
          <h2
            class="c6"
          >
            *Username Or Email
          </h2>
          <div
            class="c7"
            style="background-color: rgb(46, 52, 59); border: 2px solid rgba(87, 90, 93, 1);"
          >
            <img
              alt="auth form component username icon"
              class="c8"
              src="[object Object]"
            />
            <input
              class="c9"
              form_type="Registration"
              placeholder="*Username or Email"
              style="color: rgb(242, 242, 242);"
              type="text"
              window_height="0"
              window_width="0"
            />
            <div
              class="c10"
            >
              <img
                alt="auth form component password hide icon"
                class="c11"
                src="[object Object]"
                style="display: block;"
              />
              <img
                alt="auth form component password show icon"
                class="c12"
                src="[object Object]"
                style="display: none;"
              />
            </div>
          </div>
          <p
            class="c13"
            mode="light"
          />
        </div>
        <div
          class="c14"
          display="grid"
        >
          <h2
            class="c6"
          >
            *Username
          </h2>
          <div
            class="c7"
            style="background-color: rgb(46, 52, 59); border: 2px solid rgba(87, 90, 93, 1);"
          >
            <img
              alt="auth form component username icon"
              class="c8"
              src="[object Object]"
            />
            <input
              class="c15"
              data-testid="auth form component username input field"
              form_type="Registration"
              placeholder="*Username"
              required=""
              style="color: rgb(242, 242, 242);"
              type="text"
              window_height="0"
            />
            <div
              class="c10"
            >
              <img
                alt="auth form component password hide icon"
                class="c11"
                src="[object Object]"
                style="display: block;"
              />
              <img
                alt="auth form component password show icon"
                class="c12"
                src="[object Object]"
                style="display: none;"
              />
            </div>
          </div>
          <p
            class="c13"
            mode="light"
          />
        </div>
        <div
          class="c14"
          display="grid"
        >
          <h2
            class="c6"
          >
            *Email
          </h2>
          <div
            class="c7"
            style="background-color: rgb(46, 52, 59); border: 2px solid rgba(87, 90, 93, 1);"
          >
            <img
              alt="auth form component email icon"
              class="c16"
              src="[object Object]"
            />
            <input
              class="c17"
              form_type="Registration"
              placeholder="*Email"
              required=""
              style="color: rgb(242, 242, 242);"
              type="email"
              window_height="0"
            />
            <div
              class="c10"
            >
              <img
                alt="auth form component password hide icon"
                class="c11"
                src="[object Object]"
                style="display: block;"
              />
              <img
                alt="auth form component password show icon"
                class="c12"
                src="[object Object]"
                style="display: none;"
              />
            </div>
          </div>
          <p
            class="c13"
            mode="light"
          />
        </div>
        <div
          class="c14"
          display="grid"
        >
          <h2
            class="c6"
          >
            *Password
          </h2>
          <div
            class="c7"
            style="background-color: rgb(46, 52, 59); border: 2px solid rgba(87, 90, 93, 1);"
          >
            <img
              alt="auth form component password icon"
              class="c18"
              src="[object Object]"
            />
            <input
              class="c19"
              placeholder="*Password"
              required=""
              reveal_password="false"
              style="color: rgb(242, 242, 242);"
              type="password"
              window_height="0"
            />
            <div
              class="c20"
            >
              <img
                alt="auth form component password hide icon"
                class="c11"
                src="[object Object]"
                style="display: block;"
              />
              <img
                alt="auth form component password show icon"
                class="c12"
                src="[object Object]"
                style="display: none;"
              />
            </div>
          </div>
          <p
            class="c13"
            mode="light"
          />
        </div>
        <div
          class="c14"
          display="grid"
        >
          <h2
            class="c6"
          >
            *Confirm Password
          </h2>
          <div
            class="c7"
            style="background-color: rgb(46, 52, 59); border: 2px solid rgba(87, 90, 93, 1);"
          >
            <img
              alt="auth form component password icon"
              class="c18"
              src="[object Object]"
            />
            <input
              class="c21"
              form_type="Registration"
              placeholder="*Confirm Password"
              required=""
              reveal_password="false"
              style="color: rgb(242, 242, 242);"
              type="password"
              window_height="0"
            />
            <div
              class="c20"
            >
              <img
                alt="auth form component password hide icon"
                class="c11"
                src="[object Object]"
                style="display: block;"
              />
              <img
                alt="auth form component password show icon"
                class="c12"
                src="[object Object]"
                style="display: none;"
              />
            </div>
          </div>
          <p
            class="c13"
            mode="light"
          />
        </div>
      </div>
    </div>
    <div
      class="c22"
    >
      <button
        class="c23"
        type="submit"
      >
        Register
      </button>
      <p
        class="c24"
        style="color: rgb(242, 242, 242);"
      >
        Already a member? 
        <a
          class="c25"
          form_type="Registration"
          href="/sign-in"
        >
          Sign In
        </a>
      </p>
    </div>
  </form>
</DocumentFragment>
`;
